theseus/core/loss.py

```diff
@@ -7,8 +7,10 @@
 
 import torch
 
+_LOSS_EPS = 1e-20
```

Comment:
@mhmukadam Some comments:

- I'm OK with changing the term `Loss` for something else, but not sure what's a good alternative. Maybe `RobustLoss`? Is there another standard term for these?
- What would be good unit tests for these components? 
- I actually think that `error` and `jacobians` (unweighted) are operations that don't really make sense for this class, and we should just throw errors. Maybe we should just think of what a clear error message for these would be.  At least this is true in the way we had implemented these, which is weight first, robustify later.

Reply:
- Yes, `RobustLoss` sounds good.
- We could test the robust loss and robust cost function on some some rand values over say a th.Difference (potentially using a Lie group) and verify the Jacobians against finite difference.
- Hmm, in that case we can raise warning and say robust cost requires weighting first and return the weighted and robustified version.